src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -111,6 +116,16 @@ public void deleteCollection(String collectionName) throws PineconeException {
         }
     }
 
+    public Index createIndexConnection(String indexName) {
+        PineconeConnection connection = new PineconeConnection(config, indexName);
+        return new Index(connection);
+    }
+
+    public AsyncIndex createAsyncIndexConnection(String indexName) {
```

Comment:
Added createAsyncIndexConnection() here. The usage is described in the PR description.